src/Program.ts

```diff
@@ -846,7 +846,12 @@ export class Program {
 
         //get the completions from all scopes for this file
         let allCompletions = util.flatMap(
-            scopes.map(ctx => file.getCompletions(position, ctx)),
+            scopes.map(ctx => {
+                ctx.linkSymbolTable();
```

Comment:
The "most important" symbol checking function is `getSymbolTypeFromToken` which is currently in `BscFile` ... I'm thinking I should actually move that to `Scope`, and letting scope decide if it needs to link/unlink any symbol tables. (perhaps through some static reference to the currently linked scope)

Also, I think I should cache the result of `getSymbolTypeFromToken` in `Scope`, so that if there is subsequent call while the symbolTables are still linked with the same token, then it could just return the cached version.

What do you think?

Reply:
Since v1 is still in alpha, I'm fine with landing this as-is for now. We can address performance issues and caching strategies in future PRs (this one is big enough already...)